From 07f96302982c3874cdadabc5bed5cc90d9e8de4c Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Mon, 20 Jun 2022 16:50:11 -0500
Subject: [PATCH 1/1] phosphor-pid-control : Modify Sensor based Temp
 simulation added

Temperature simulation file changed to sensor name based file.
e.g.  "/etc/thermal.d/P0_DIMM5_Temp"

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 pid/zone.cpp | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/pid/zone.cpp b/pid/zone.cpp
index 16be5db..1ba15eb 100644
--- a/pid/zone.cpp
+++ b/pid/zone.cpp
@@ -35,6 +35,8 @@
 #include <string>
 #include <filesystem>

+#define TEMP_FILE_PATH "/etc/thermal.d/"
+
 using tstamp = std::chrono::high_resolution_clock::time_point;
 using namespace std::literals::chrono_literals;

@@ -358,10 +360,10 @@ void DbusPidZone::updateFanTelemetry(void)
     return;
 }

-static double fileParseTemp(void)
+static double fileParseTemp(const std::string& fileName)
 {
     std::string errText;
-    std::string fileName = "/etc/thermal.d/temp";
+
     try
     {
         std::ifstream ifs;
@@ -400,9 +402,11 @@ void DbusPidZone::updateSensors(void)
         /* Following code added for simulating the CPU Temp using /etc/thermal.d/temp file.
          * If /etc/thermal.d/temp file exists then swampd will read temp value from it
         */
-        if ( std::filesystem::exists("/etc/thermal.d/temp") )
+        /* read  Temp value from simulation file */
+        std::string fileName = std::string(TEMP_FILE_PATH) + sensor->getName();
+        if ( std::filesystem::exists(fileName))
         {
-            temp_val = fileParseTemp();
+            temp_val = fileParseTemp(fileName);
             r.value = temp_val;
         }

--
2.17.1

